File: serveur/app/models/actual_workout.py
# Exemple : app/models/actual_workout.py

from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Boolean, func
from app.database import Base

import uuid

class ActualWorkout(Base):
    __tablename__ = "actual_workouts"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    name = Column(String, nullable=False)
    date = Column(String, nullable=False)   # au format ISO : YYYY-MM-DD
    notes = Column(String, nullable=True)
    location = Column(String, nullable=True)
    is_done = Column(Boolean, default=False, nullable=False)

    updated_at = Column(DateTime(timezone=True), nullable=True)

    uuid = Column(String, default=lambda: str(uuid.uuid4()), nullable=False, unique=True)

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/ActualWorkout.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import com.google.gson.annotations.SerializedName
import java.util.UUID

@Entity(
    tableName = "actual_workouts",
    indices = [
        androidx.room.Index(value = ["uuid"], unique = true),
        androidx.room.Index(value = ["date"]),
    ]
)
data class ActualWorkout(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,
    @ColumnInfo(name = "name") val name: String,
    val date: String,   // date au format ISO8601
    val notes: String? = null,
    val location: String? = null,
    @ColumnInfo(name = "is_done")val isDone: Boolean = false,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
